mi/dataset/parser/suna.py

```diff
@@ -0,0 +1,215 @@
+
+    @staticmethod
+    def _date_time_sample_values_to_ntp_timestamp(date_sample_str, time_sample_str):
+
+        year = int(date_sample_str[0:4])
+        days = int(date_sample_str[4:7])
+
+        hours_float = float(time_sample_str)
+
+        date_time_val = datetime.datetime(year, 1, 1) + datetime.timedelta(days=days-1, hours=hours_float)
+        ntp_timestamp = ntplib.system_to_ntp_time(calendar.timegm(date_time_val.timetuple()))
+
+        return ntp_timestamp
+
+    def parse_file(self):
+        for line in self._stream_handle:
+
+            # DCL/Telemetered
+            if not line.startswith('SATSLF') and 'SATSLF' in line or 'SATNDF' in line:
+                raw_data = line.split(',')
+                if len(raw_data) != self._raw_data_length:
+                    continue
+
+                particle_class = SunaDclRecoveredDataParticle
+
+                # Get rid of the date and time at the beginning of the line. raw_data will start with SATSLF...
+                # And then split it once more to get a list of the data
+                raw_data = line.split(' ', 2)[2].split(',')
+
+                date_time = raw_data[1:3]
+                timestamp = self._date_time_sample_values_to_ntp_timestamp(
+                    date_time[0],
+                    date_time[1],
+                )
+
+                raw_data.insert(1, raw_data[0][3:6])
+                raw_data[0] = raw_data[0][0:3]
+                particle = self._extract_sample(particle_class, None, raw_data,
+                                                internal_timestamp=timestamp)
+
+                self._record_buffer.append(particle)
+
+            elif line.startswith('SATSLF') or line.startswith('SATSDF'):
+                raw_data = line.split(',')
+                if len(raw_data) != self._raw_data_length:
+                    continue
+
+                particle_class = SunaInstrumentRecoveredDataParticle
+                raw_data.insert(1, raw_data[0][3:6])
+                raw_data[0] = raw_data[0][0:3]
+
+                particle = self._extract_sample(particle_class, None, raw_data)
+                self._record_buffer.append(particle)
+
+    def get_records(self, num_records_requested=1):
+        """
+        Returns a list of particles that is  equal to the num_records_requested when there are that many particles
+        are available or a list of particles less than the num_records_requested when there are fewer than
+        num_records_requested available.
+        """
+        particles_to_return = []
+
+        if num_records_requested:
+
+            # If the file was not read, let's parse it
+            if not self._file_parsed:
+                self.parse_file()
+
+            # Iterate through the particles returned, and pop them off from the beginning of the record
+            # buffer to the end
+            while len(particles_to_return) is not num_records_requested and len(self._record_buffer) > 0:
```

Comment:
if the length of an array is 0, its boolean value is `False`:
```python
while len(particles_to_return) is not num_records_requested and self._record_buffer:
```
Look how pythonic it looks!